src/Ractive/prototype/set.js

```diff
@@ -10,9 +10,9 @@ export default function Ractive$set ( keypath, value ) {
 	if ( isObject( keypath ) ) {
 		const map = keypath;
 
-		for ( keypath in map ) {
-			if ( map.hasOwnProperty( keypath) ) {
-				set( this, keypath, map[ keypath ] );
+		for ( const k in map ) {
+			if ( map.hasOwnProperty( k) ) {
+				set( this, k, map[k] );
 			}
```

Comment:
put this at the top of your main file to see for yourself:

```js
Array.prototype.forEach = function (fn) {
  for (var i = 0; i < this.length; i++) {
    fn(this[i], i)
  }
}
```

additionally converting to a normal for-loop will reduce the number of functions needed drastically, further improving performance.